hs_core/templates/search/search.html

```diff
@@ -106,134 +95,398 @@ <h4 class="panel-title">
+                if (event.which == 13) {
+                    event.preventDefault();
+                    localStorage.clear();
+                    $("#search-field").submit();
+                }
+            });
+            /**
+             * temporarily keep the following functions for checkboxes
+             */
+            $(".faceted-selections").click(function () {
+                var tokens = $(this).val().split(",");
+                var key = tokens[0];
+                var value = tokens[1];
+                var textSearch = $("#id_q").val();
+                var searchURL = "?q=" + textSearch;
+                var windowPath = window.location.pathname;
+                var requestURL =  windowPath + searchURL;
+
+                if($(this).is(":checked")) {
+                    requestURL = requestURL + "&selected_facets="+key+"_exact:"+value
+                    $(".faceted-selections").each(function () {
+                        var val = localStorage[$(this).attr("id")];
+                        //var isChecked = val !== undefined ? val == "true" : false;
+                        var isChecked = val === 'true';
+                        if(isChecked){
+                            var arr = $(this).val().split(",");
+                            var key = arr[0];
+                            var value = arr[1];
+                            requestURL += "&selected_facets="+key+"_exact:"+value;
+                        }
+                    });
+                    window.location = requestURL;
+                    localStorage[$(this).attr("id")] = this.checked;
+                }else{
+                    localStorage.removeItem($(this).attr("id"));
+                    var textSearch = $("#id_q").val();
+                    var updateURL = "/search/?q=" + textSearch;
+                    $(".faceted-selections").each(function () {
+                        var val = localStorage[$(this).attr("id")];
+                        //var isChecked = val !== undefined ? val == "true" : false;
+                        var isChecked = val === 'true';
+                        if(isChecked){
+                            var arr = $(this).val().split(",");
+                            var key = arr[0];
+                            var value = arr[1];
+                            updateURL += "&selected_facets="+key+"_exact:"+value;
+                        }
+                    });
+                    window.location = updateURL;
+                }
+            });
+
             $(".faceted-selections").each(function () {
                 var val = localStorage[$(this).attr("id")];
                 var isChecked = val !== undefined ? val == "true" : false;
                 $(this).prop("checked", isChecked);
             });
-        $("#faceted-search").click(function(){
-            localStorage.clear();
+
+            $("#faceted-search").click(function(){
+                localStorage.clear();
```

Comment:
You need to *only* clear keys you put in `localStorage` because this `clear()` will remove keys that may have been put there by other parts of the site for other reasons, breaking things.